### PR TITLE
API Milestone 1

### PR DESCRIPTION
**Pull request by big-kahuna-burger**
Following routes are completed with this PR:
GET /application
GET /application/{applicationId}
POST /application
POST /application/{applicationId}/submit
POST /application/{applicationId}/watch

This has already too much commits, so I'm thinking to approach it a bit more granular

### CODE REVIEW

**Comment by big-kahuna-burger** [comment]
Missing setting source of the application impl/test. (admin/user)

File: mocks/dummy.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+//user mock
```
Comment (amirsabersharifi) [inline mocks/dummy.js]: for future you can use fake.js to autogenerate names and stuff like this

File: modules/application.js [hunk 2]
```diff
@@ -1,75 +1,104 @@
...
+        method: 'GET',
+        json: true,
+        resolveWithFullResponse: true,
+        headers:{
+            'Prefer': 'return=representation'
+        }
+    });
 };
-exports.addWatch = function (applicationId, watchId) {
+exports.getOne = function (filter) {
```
Comment (amirsabersharifi) [inline modules/application.js]: I don't like the name of this function

File: package.json [hunk 3]
```diff
@@ -22,5 +22,9 @@
   "repository": {
     "type": "git",
     "url": "git+https://github.com/Vyrent/vyrent_env"
+  },
+  "devDependencies": {
+    "eslint": "^3.7.1",
+    "jshint": "^2.9.3"
```
Comment (amirsabersharifi) [inline package.json]: is it necessary to have both? I think eslint is good enought

File: test/lead.test.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+var dummyVendor;
+var dummyWatch;
+var dummyWatch2;
+var dummyLead;
+var dummyLead2;
+var initialCount;
+
+describe('Lead', function () {
+    before(function (done) {
+        User.createUserWithExternalId(dummy.userProfile().user_id)
```
Comment (amirsabersharifi) [inline test/lead.test.js]: For later, there can be a fixture configuration to prepopulate database to a point that has all possible situations and then test from there with needed behavrious

File: modules/api/applications/createApplications.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+"use strict";
+var Promise = require('bluebird');
+var Application = require('../../application');
+var Auth = require('../../auth');
+var Util = require('../../util');
+
+exports.handler = (event, context, callback) => {
```
Comment (amirsabersharifi) [inline modules/api/applications/createApplications.js]: move this part to index.js and just do the business logic in api files. then in index.js files(yes there will be one for each api call) we convert lambda events to our data

**Comment by big-kahuna-burger** [comment]
```100 passing (966ms)```

**Pull request merged**